intro-HPC/ch_openFOAM.tex

```diff
@@ -0,0 +1,319 @@
+rather than the standard \lstinline|mpirun| command;
+see \autoref{sec:best-practices-mympirun} for more information on \lstinline|mympirun|.
+
+To use \lstinline|mympirun|, make sure that the \lstinline|vsc-mympirun| module is loaded.
+
+\begin{prompt}
+%\shellcmd{module load vsc-mympirun}%
+\end{prompt}
+
+Note that you should \emph{not} specify a specific version here,
+see also \autoref{sec:best-practices-mympirun-module}.
+
+To pass down the environment variables required to run OpenFOAM (which were defined by the
+\lstinline|$FOAM_BASH| script, see \autoref{sec:best-practices-openfoam-environment})
+to each of the MPI processes used in a parallel OpenFOAM execution,
+the \lstinline|$MYMPIRUN_VARIABLESPREFIX| environment variable must be defined as follows,
+prior to running the OpenFOAM simulation with \lstinline|mympirun|:
+
+\begin{prompt}
+%\shellcmd{export MYMPIRUN\_VARIABLESPREFIX=WM\_PROJECT,FOAM,MPI}%
+\end{prompt}
+
+Whenever you are instructed to use a command like \lstinline|mpirun -np <N>| ...,
+use \lstinline|mympirun| ... instead; \lstinline|mympirun| will automatically detect the number of
+processor cores that are available (see also \autoref{sec:best-practices-mympirun-cores}).
+
+\subsection{Domain decomposition and number of processor cores}
+\label{sec:best-practices-openfoam-domain-decomposition-processor-cores}
+
+To run OpenFOAM in parallel, you must decompose the domain into multiple subdomains.
+Each subdomain will be processed by OpenFOAM on one processor core.
+
+Since \lstinline|mympirun| will automatically use all available cores, you need to make sure
+that the number of subdomains matches the number of processor cores that will be used by \lstinline|mympirun|.
+If not, you may run into an error message like:
+
+\begin{prompt}
+number of processor directories = 4 is not equal to the number of processors = 16
+\end{prompt}
+
+In this case, the case was decomposed in 4 subdomains, while the OpenFOAM simulation was started with 16 processes
+through \lstinline|mympirun|.
+To match the number of subdomains and the number of processor cores used by \lstinline|mympirun|,
+you should either:
+
+\begin{itemize}
+\item adjust the value for \lstinline|numberOfSubdomains| in \lstinline|system/decomposeParDict|
+    (and adjust the value for \lstinline|n| accordingly in the domain decomposition coefficients),
+    and run \lstinline|decomposePar| again; or
+\item submit your job requesting exactly the same number of processor cores as there are subdomains (see the
+    number of \lstinline|processor*| directories that were created by \lstinline|decomposePar|)
+\end{itemize}
+
+Alternatively you can specify to \texttt{mympirun} that it should only start a certain number of MPI processes,
+via \lstinline|--hybrid| (see \autoref{sec:best-practices-mympirun-hybrid}),
+taking into account the number of requested workernodes,
+or \lstinline|--universe| (see \autoref{sec:best-practices-mympirun-universe}).
```

Comment:
refer to other options section here (don't document `--universe` in the `mympirun` section)